firmware/application/apps/ui_fileman.cpp

```diff
@@ -267,14 +267,15 @@ void FileManagerView::on_refactor(NavigationView& nav) {
 		std::string destination_path = current_path.string();
 		if (destination_path.back() != '/')
 			destination_path += '/';
-		destination_path = destination_path + buffer;
+
+		destination_path = get_selected_path().string().back() != '/' ? destination_path + buffer + extension_buffer : destination_path + buffer;
 
 		rename_file(get_selected_path(), destination_path);  //rename the selected file
 
 		auto selected_path = get_selected_path();
```

Comment:
shouldn't we use destination_path here since we renamed, but not reloaded via load_directory_content yet ?

Reply:
> shouldn't we use destination_path here since we renamed, but not reloaded via load_directory_content yet ?

yes, im so sorry....